int32 num1
int32 num2
int32 num3
---
int32 product
